.github/workflows/consuming.yml

```
@@ -89,12 +92,16 @@ jobs:
         uses: actions/checkout@v2
         with:
           repository: submariner-io/${{ matrix.project }}
+          path: ${{ matrix.project }}
+
+      - name: Copy Dockerfile.dapper if necessary
+        run: test -f ${{ matrix.project }}/Dockerfile.dapper || cp Dockerfile.dapper ${{ matrix.project }}/
 
       - name: Make sure ${{ matrix.project }} is using the built Shipyard image
-        run: sed -i 's/shipyard-dapper-base:*.*/shipyard-dapper-base:dev/' Dockerfile.dapper
+        run: sed -i 's/shipyard-dapper-base:*.*/shipyard-dapper-base:dev/' ${{ matrix.project }}/Dockerfile.dapper
```

Why not make this conditional instead?
```suggestion
        run: [ ! -f  ${{ matrix.project }}/Dockerfile.dapper ] || sed -i 's/shipyard-dapper-base:*.*/shipyard-dapper-base:dev/' ${{ matrix.project }}/Dockerfile.dapper
```

?

---

Because we want to push our `Dockerfile.dapper` into the project being tested, not rely on whatever version is downloaded — the latter gets two things wrong: it ends up relying on the `devel` image, not the new Shipyard image, and it uses the `devel` `Dockerfile.dapper`, not the one supposedly being tested.

---

Put another way, this line should never be conditional because we always want to replace the tag.

---

Ah, right, got it.